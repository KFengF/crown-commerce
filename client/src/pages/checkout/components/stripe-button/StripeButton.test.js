import React from 'react';
import { shallow } from 'enzyme';
import StripeButton from './StripeButton.component';

describe('StripeButton component', () => {
    let wrapper;
    let mockPrice;

    beforeEach(() => {
        mockPrice = 50;
        const mockProps = {
            price: mockPrice
        }
        wrapper = shallow(<StripeButton { ...mockProps } />);
    });

    it('should render StripeButton component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('price', () => {
        it('should print Your total is $50 in StripeCheckout component description', () => {
            expect(wrapper.find('ReactStripeCheckout').prop('description'))
                .toBe('Your total is $' + mockPrice);
        });
    });
});