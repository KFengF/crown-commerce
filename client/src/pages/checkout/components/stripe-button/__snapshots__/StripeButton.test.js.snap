// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StripeButton component should render StripeButton component 1`] = `ShallowWrapper {}`;
